File: onebot-client/src/main/java/cn/evole/onebot/sdk/response/group/ForwardMsgResp.java
package cn.evole.onebot.sdk.response.group;

import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.List;

/**
 * @author MrXiaoM
 */
@Data
public class ForwardMsgResp {
    /**
     * 消息节点列表
     */
    @SerializedName("message")
    private List<Node> message;
    /**
     * 消息节点
     */
    @Data
    public static class Node {

        @SerializedName("user_id")
        private String userId;

        @SerializedName("nickname")
        private String nickname;

        /**
         * 消息内容
         */
        @SerializedName("content")
        private String content;
    }
}

File: overflow-core/src/main/kotlin/cn/evolvefield/onebot/sdk/response/group/ForwardMsgResp.kt
package cn.evolvefield.onebot.sdk.response.group

import cn.evolvefield.onebot.sdk.util.json.ForwardMsgAdapter
import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName

/**
 * @author MrXiaoM
 */
@JsonAdapter(ForwardMsgAdapter::class)
class ForwardMsgResp {
    /**
     * 消息节点列表
     */
    var message: List<Node> = mutableListOf()
    /**
     * 消息节点
     */
    class Node {
        var time = 0
        var messageType = ""
        var messageId = 0
        var realId = 0
        var peerId = 0L
        var targetId = 0L
        var sender: Sender? = null
        var message = ""
    }

    /**
     * 消息发送者
     */
    class Sender {
        @SerializedName("user_id")
        var userId = 0L
        @SerializedName("nickname")
        var nickname = ""
        @SerializedName("sex")
        var sex = ""
        @SerializedName("age")
        var age = 0
        @SerializedName("uid")
        var uid = ""
    }
}
